client/src/components/AdminPopup/AdminPopup.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { forwardRef } from 'react';
+import RoleSpecificFields from './RoleSpecificFields';
+import RoleSelector from './RoleSelector';
+import {
+  useAddUserMutation,
+  useDepartmentsAndGendersAndRolesQuery
+} from '../../generated/graphql-types';
+import { useCreateUserForm } from './useCreateUserForm';
+import { AdminPopupProps, InputError } from './AdminPopup.types';
+
+const AdminPopup = forwardRef(({ close }: AdminPopupProps, ref) => {
```

Comment:
J'ai creusé et je pense avoir compris l'histoire de forwardRef : 
- si on utilise l'attribut ref on doit l'utiliser
- si on passe la ref en props ("dialogRef" par exemple) on en a pas besoin, par ce que la ref "reste dans react", j'ai fait une démo dans ce commit : https://github.com/WildCodeSchool-CDA-FT-2024-09/JS-CDA-Projet-2-Team-C/commit/f936adc0dc057be2c57403a1ec6331d7f2ab5afe  (et ca règle même une erreur TS )

Qu'on merge ca ou autre chose m'est égal, mais je viens d'apprendre un truc, donc merci !

Reply:
T'as vérifié ta console du navigateur pendant ton test?
Sinon bizarre, car de mon coté j'avais une erreur explicite comme quoi il fallait utiliser forwardRef

Et sinon par rapport à l'utilisation de useRef en elle même dans le contexte de react:
<img width="617" alt="image" src="https://github.com/user-attachments/assets/1b21173c-638b-493d-a656-69939030ee67" />

Reply:
J'ai switché sur ton commit pour voir, et il n'y a pas d'erreur justement car c'est passé en prop, mais du coup j'ai vérifié quelle était la différence entre la prop et la ref, et la prop semble etre une mauvaise pratique:

<img width="589" alt="image" src="https://github.com/user-attachments/assets/0798bcc0-722b-4f04-8673-96f4fa57b382" />

et

<img width="635" alt="image" src="https://github.com/user-attachments/assets/23a509c7-be2e-439b-bab5-838deff3a7ed" />

Bref merci ChatGPT, et c'est intéressant d'aller profondément dans les choses

Sinon de mon coté pour moi l'avantage de useRef c'est que ça trigger pas de rerender, donc je me demande si de passer par des prop ça pourrait pas aussi faire des soucis de ce coté, mais je m'y connais pas assez

Reply:
La doc officielle semble être d'accord avec toi : https://react.dev/learn/manipulating-the-dom-with-refs#accessing-another-components-dom-nodes